lahja/base.py

```diff
@@ -248,3 +241,86 @@ def __reduce__(self) -> None:  # type: ignore
         event = await agen.asend(None)
         await agen.aclose()
         return event
+
+    def is_remote_subscribed_to(
+        self, remote_endpoint: str, event_type: Type[BaseEvent]
+    ) -> bool:
+        """
+        Return ``True`` if the specified remote endpoint is subscribed to the specified event type
+        from this endpoint. Otherwise return ``False``.
+        """
+        for endpoint, subscriptions in self.get_connected_endpoints_and_subscriptions():
+            if endpoint != remote_endpoint:
+                continue
+
+            for subscribed_event_type in subscriptions:
+                if subscribed_event_type is event_type:
+                    return True
+
+        return False
```

Comment:
Wondering if we should error in this case (that no endpoint name matched the provided name).  Seems like an easy mistake to make.

Reply:
I was wondering the same but here is why I didn't do that:

1. If we look at the `wait_until_*` family of APIs, I think it is totally reasonable to start waiting on remote subscriptions even *before* a specific connection has been established. I'm trying to see this from a broader non-trinity based perspective but even *within* Trinity I can see that being useful. E.g. Hot-plugin replacement/loading could cause new endpoints to come up later in the lifecycle of the application and having this raise an exception may make that harder.

2. Of course, we could say we change `is_remote_subscribed_to` to raise an exception and then have the `wait_until_*` APIs catch and ignore this so that only the `is_remote_subscribed_to` API raises but I think that would introduce an inconsistency between these APIs with little gain in return.